compiler/catala_utils/message.mli

```diff
@@ -66,6 +66,17 @@ end
 exception CompilerError of Content.t
 exception CompilerErrors of Content.t list
 
+type error_kind = Lexing | Parsing | Typing | Generic
+
+type error = {
+  kind : error_kind;
+  msg : string;
+  pos : Pos.t option;
+  suggestion : string list option;
+}
+
+val register_error_notifier : (error -> unit) -> unit
```

Comment:
So if I understand correctly this `error` type is the one expected by the LSP for an error to be displayed in an IDE ? If so, it might be renamed into `lsp_error` and the function to `register_lsp_error_notifier`.

Reply:
It is not particularly meant to be specific to LSP. Although, as of right now, I don't see any other immediate use-cases. I'm fine renaming it if you'd like.

Reply:
Changed `error` into `lsp_error`